File: examples/eApi v1.9/java/eAPI-v19-connector/src/main/java/cz/monetplus/mips/eapi/v19/connector/entity/responses/ButtonInitResponse.java
package cz.monetplus.mips.eapi.v19.connector.entity.responses;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import cz.monetplus.mips.eapi.v19.connector.entity.SignBase;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@EqualsAndHashCode(callSuper = false)
@Data
@AllArgsConstructor @NoArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class ButtonInitResponse extends SignBase {
    private String payId;
    private Integer resultCode;
    private String resultMessage;
    private Integer paymentStatus;
    private Endpoint redirect;

    @Override
    public String toSign() {
        StringBuilder sb = new StringBuilder();
        add(sb, payId);
        add(sb, dttm);
        add(sb, resultCode);
        add(sb, resultMessage);
        add(sb, paymentStatus);
        if(null != redirect) add(sb, redirect.toSign());
        deleteLast(sb);
        return sb.toString();
    }
}

File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/Response/ButtonInitResponse.cs
using System.Text;

namespace CsobGatewayClientExample.Communication.DataObjects.Response;

public class ButtonInitResponse : SignBase
{
    public string? PayId { get; set; }
    public int ResultCode { get; set; }
    public string? ResultMessage { get; set; }
    public long PaymentStatus { get; set; }
    public RedirectRes? Redirect { get; set; }

    public override string ToSign()
    {
        var sb = new StringBuilder();
        Add(sb, PayId);
        Add(sb, Dttm);
        Add(sb, ResultCode);
        Add(sb, ResultMessage);
        Add(sb, PaymentStatus);
        DeleteLast(sb);
        return sb.ToString();
    }
}